assertion_test.go

```diff
@@ -477,3 +477,183 @@ func TestAssertion_Strings(t *testing.T) {
 		}
 	})
 }
+
+func TestAssertion_ValidateTraits(t *testing.T) {
```

Comment:
Let's move this test right above TestAssertion_Validation. And also let's rename TestAssertion_Validation to TestAssertion_ValidateAssertion.